chocon.go

```diff
@@ -104,7 +106,14 @@ Compiler: %s %s
 			runtime.Compiler,
 			runtime.Version())
 		return
+	}
 
+	upstreamURL := new(url.URL)
+	if opts.Upstream != "" {
+		upstreamURL, err = url.Parse(opts.Upstream)
```

Comment:
`url.Parse()`'s behavior is a little ambiguous. For example, the values below are valid.

1. http://example.com
1. https://example.com
1. example.com

But  in this PR looks to expect 1 or 2 instead of 3. So, houw about adding the condition below?

```
strings.HasPrefix(server, "http://") || strings.HasPrefix(server, "https://")
```

Reply:
Added schema and host check